chalice/utils.py

```diff
@@ -8,13 +8,15 @@
 import shutil
 import sys
 import tarfile
+from datetime import datetime, timedelta
 import subprocess
 
 import click
 from typing import IO, Dict, List, Any, Tuple, Iterator, BinaryIO, Text  # noqa
 from typing import Optional, Union  # noqa
-from typing import MutableMapping  # noqa
+from typing import MutableMapping, Callable  # noqa
 from typing import cast  # noqa
+from botocore.utils import parse_timestamp
```

Comment:
I'd probably say we should vendor this function or write a subset of it chalice. There is definitely no guarantees we break this in the future.

Reply:
Yeah initially I was going to just vendor it, but then I noticed you all added some recent fixes for windows (which also makes it a little more tedious to backport).  I'd probably miss those updates, but I suppose that's better than breaking.